TangoRosStreamer/app/src/main/java/eu/intermodalics/tango_ros_streamer/RunningActivity.java

```diff
@@ -346,6 +351,21 @@ private void getTangoPermission(String permissionType) {
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
+        mRestartTangoAlertReceiver = new BroadcastReceiver() {
+            @Override
+            public void onReceive(Context context, Intent intent) {
+                mTangoServiceClientNode.callTangoConnectService(TangoConnectRequest.DISCONNECT);
+                mParameterNode.uploadPreferencesToParameterServer();
+                mCreateNewMap = mSharedPref.getBoolean(getString(R.string.pref_create_new_map_key), false);
+                if (mCreateNewMap) {
+                    mSaveButton.setVisibility(View.VISIBLE);
+                } else {
+                    mSaveButton.setVisibility(View.INVISIBLE);
```

Comment:
Using `View.GONE` is slightly better because the button does not take up any memory space in this case.